### PR TITLE
docs: Clarify expectations of the mentor/mentee relationship.

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>

Fixes #TODO

### CODE REVIEW

File: docs/mentoring.md [hunk 1]
```diff
@@ -50,5 +57,4 @@ For program specific questions, please refer to [GSoC FAQ](https://developers.go
 If you have any problems with your relationship, please contact us:
 
 * alex_collins@intuit.com
-<!-- * TODO - if some want to get help with Alex, then it cannot be alex they email -->
-
+* terrytangyuan@gmail.com
```
Comment (alexec) [inline docs/mentoring.md:60]: @terrytangyuan ?

Comment (terrytangyuan) [reply docs/mentoring.md:60]: This works.

File: .github/ISSUE_TEMPLATE/mentoring_request.md [hunk 2]
```diff
@@ -15,3 +15,7 @@ Answer: **I have previous experience in React.**
 **Is there any particular issue you'd like to work on? You may want to check out the [list of good first issues](https://github.com/argoproj/argo-workflows/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).**
 
 Answer: **I would like to work with issue #7920**
+
+**Do you want to work with your mentor async (i.e. by Slack messaging or issue comments), or sync (e.g. using Zoom video)?** Please include your time-zone if you want to work sync.
```
Comment (terrytangyuan) [inline .github/ISSUE_TEMPLATE/mentoring_request.md:19]: Do we need Slack usernames for each mentor?

File: docs/mentoring.md [hunk 3]
```diff
@@ -13,14 +13,25 @@ Mentors and mentees must abide by the [code of conduct](https://github.com/cncf/
...
 
-Create [a mentoring request](https://github.com/argoproj/argo-workflows/issues/new/choose) with the title "I would like a mentor" and answer the questions in the issue template.
-Note that this is only for general mentoring. For Google Summer of Code, please **do not** create a mentoring request and find the instructions in the relevant section below.
+We establish and end mentor/mentee relationships using issues. The mentee creates the issue and the mentor is assigned the issue. The relationship ends when the issue is closed.
 
-We'll find the right mentor for you.
+To set expectations:
+
+1. Create [a mentoring request](https://github.com/argoproj/argo-workflows/issues/new/choose) with the title "I would like a mentor" and answer the questions in the template.
+2. A mentor will comment offering to mentor you. It may take a few days for a mentor to respond.
```
Comment (terrytangyuan) [inline docs/mentoring.md:21]: +1 this would reduce frustration

File: docs/mentoring.md [hunk 4]
```diff
@@ -13,14 +13,25 @@ Mentors and mentees must abide by the [code of conduct](https://github.com/cncf/
...
 ## How To Get A Mentor
 
-Create [a mentoring request](https://github.com/argoproj/argo-workflows/issues/new/choose) with the title "I would like a mentor" and answer the questions in the issue template.
-Note that this is only for general mentoring. For Google Summer of Code, please **do not** create a mentoring request and find the instructions in the relevant section below.
+We establish and end mentor/mentee relationships using issues. The mentee creates the issue and the mentor is assigned the issue. The relationship ends when the issue is closed.
 
-We'll find the right mentor for you.
+To set expectations:
+
+1. Create [a mentoring request](https://github.com/argoproj/argo-workflows/issues/new/choose) with the title "I would like a mentor" and answer the questions in the template.
```
Comment (terrytangyuan) [inline docs/mentoring.md:20]: We can also use this link? https://github.com/argoproj/argo-workflows/issues/new?assignees=&labels=mentoring%2Ctriage&template=mentoring_request.md

**Review by terrytangyuan — APPROVED** [review]
Minor comments. Feel free to merge as is.

File: .github/ISSUE_TEMPLATE/mentoring_request.md (hunk 2 above)
Comment (alexec) [reply .github/ISSUE_TEMPLATE/mentoring_request.md:19]: not sure, I've not needed it yet

**Pull request merged**